python/dask_cudf/dask_cudf/tests/test_groupby.py

```diff
@@ -82,28 +84,29 @@ def test_groupby_cumulative(aggregation, pdf, series):
     gdf = cudf.DataFrame.from_pandas(pdf)
     ddf = dask_cudf.from_cudf(gdf, npartitions=5)
 
-    if pdf.isna().sum().any():
-        with pytest.xfail(
-            reason="https://github.com/rapidsai/cudf/issues/12055"
-        ):
-            gdf_grouped = gdf.groupby("xx")
-    else:
-        gdf_grouped = gdf.groupby("xx")
-
+    gdf_grouped = gdf.groupby("xx")
     ddf_grouped = ddf.groupby("xx")
 
     if series:
         gdf_grouped = gdf_grouped.xx
         ddf_grouped = ddf_grouped.xx
 
-    a = getattr(gdf_grouped, aggregation)()
-    b = getattr(ddf_grouped, aggregation)()
-
-    if aggregation == "cumsum" and series:
-        with pytest.xfail(reason="https://github.com/dask/dask/issues/9313"):
-            dd.assert_eq(a, b)
+    if pdf.isna().sum().any():
+        # https://github.com/rapidsai/cudf/issues/12055
+        gdf_grouped = gdf.groupby("xx")
+        context = pytest.raises(ValueError)
     else:
-        dd.assert_eq(a, b)
+        context = contextlib.nullcontext()
+    with context:
+        a = getattr(gdf_grouped, aggregation)()
+        b = getattr(ddf_grouped, aggregation)()
+
+        if aggregation == "cumsum" and series:
+            with pytest.raises(AssertionError):
+                # https://github.com/dask/dask/issues/9313
+                dd.assert_eq(a, b)
+        else:
+            dd.assert_eq(a, b)
```

Comment:
I'll go round again and do this properly as part of fixing up the other packages.